src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ProjectState.cs

```diff
@@ -84,7 +92,10 @@ public static ProjectState Create(HostWorkspaceServices services, HostProject ho
         }
 
         // Internal set for testing.
-        public IReadOnlyDictionary<string, DocumentState> Documents { get; internal set; }
+        public ImmutableDictionary<string, DocumentState> Documents { get; internal set; }
+
+        // Internal set for testing.
+        public ImmutableDictionary<string, ImmutableArray<string>> ImportsToIncludingDocuments { get; internal set; }
```

Comment:
Will rename this to `ImportsToRelatedDocuments` - that's a name I thought of while writing the part after this.